crates/bitcoin/src/types.rs

```diff
@@ -24,6 +24,13 @@ use serde::{Deserialize, Serialize};
 
 pub(crate) const SERIALIZE_TRANSACTION_NO_WITNESS: i32 = 0x4000_0000;
 
+#[derive(Encode, Decode, Clone, TypeInfo, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct FullTransactionProof {
```

Comment:
Can you provide some docs why this is required? i.e. having both the user and coinbase tx